### PR TITLE
✨ OSIDB-1204 Create CVSSv3 calculator

### PR DESCRIPTION
**Pull request by C-Valen**
## Checklist:

- [x] Linting passed _(there are warns not related to this branch)_
- [x] Type checks passed
- [x] Tests suite passed
- [x] Commits consolidated
- [x] Changelog updated
- [x] Test cases added/updated
- [x] Jira ticket updated

## Summary:

Implements Cvss calculator, using Cvss v3.1 [specification](https://www.first.org/cvss/v3.1/specification-document),
as an embedded component on `FlawForm` and main controller for `Cvss3 Vector` and `Cvss3 Score` fields.

**Features**:
- Dynamic collapsible behavior based on Cvss3 vector input focus
- Compact calculator style inspired by [cvss.js.org](https://cvss.js.org/)
- Dynamic colors on Cvss3 factors based on severity
- Validation for incomplete Cvss3 vector
- Erase button to cleanup values on Cvss3 vector and score
- Factor tooltips with description of each factor value
- Handles pasting in CVSS scores: vector order does not matter! s/o to @C-Valen for his implementation which supports this (added by @superbuggy) 
- Highlights sections of CVSS when editing (added by @superbuggy)
- Dynamically calculates color values based on severity (added by @superbuggy) 

## Changes:

- Creates embedded component `CvssCalculator` on `FlawForm`
- Removes deprecated code related to previous Cvss calculator link from `FlawForm`
- Creates a composable `useCvssCalculator` with calculator logic and utilities
- Updates test cases on `FlawForm` for `CvssCalculator`
- Creates `CvssCalculator` specific test cases
- Fixes minor bugs on `LabelStatic`

## Considerations:

- Cvss vector and score fields are now wrapped inside Cvss calculator component
- Cvss vector and score fields are now only editable by calculator actions
- Cvss vector and score fields are now both highlighted in yellow when empty
- Cvss vector validation is handled on useCvssCalculator composable

### CODE REVIEW

**Comment by C-Valen** [comment]
Notes from Spring Review on Cvss Calculator:
- Improve contrast ratio on cvss vector when showing colors
- Check for possible conflicts on OSIDB score calculation
- Display score as integer when there are not decimals
- Allowing users to paste custom values on cvss vector input? (Discussion)

**Comment by C-Valen** [comment]
> Notes from Spring Review on Cvss Calculator:
> ...

Updates:
- Improved contrast ratio generally
- Verified that score is displayed as integer when there are not decimals

image

File: src/components/widgets/LabelStatic.vue [hunk 1]
```diff
@@ -38,6 +44,10 @@ defineProps<{
   word-wrap: break-word;
 }
 
+span.form-control.alert {
+  padding-block: 10px;
+}
+
 .osim-static-label .osim-static-label-top-style .top-label {
   display: inline-block;
   background-color: #DEE2E6;
```
Comment (superbuggy) [inline src/components/widgets/LabelStatic.vue:53]: 🙏

**Comment by superbuggy** [comment]
https://github.com/RedHatProductSecurity/osim/assets/17716232/735f441f-97ce-4d7c-88d1-88ba8600c910

**Comment by C-Valen** [comment]
@superbuggy the pasting vector feature looks nice, good job! 
It looks clean to me to be merged, feel free to do the honors once we get the reviews.

I just add couple ideas to may be consider/discuss for the future:

- Allow "undo/redo" (for example to recover last value after pasting a string by mistake)
- Make blue hovering on button instead factor header +  adding previewing hovered button factor value on vector string

**Review by tboyetteredhat — APPROVED** [review]
LGTM. Nice job 🚀

**Pull request merged**